### ISSUE TITLE
Locked out of Ford Account

### ISSUE THREAD

**Issue by Madbeefer** [itchannel/fordpass-ha#214]
I wasn't able to access my ford account at all. I finally got a hold of someone and they said they lock any account that has over 20,000 API calls in a week. I didn't see anywhere to change the interval for API calls.

**Comment by AlteVerwischteZiege** [comment]
What else do you have that touches your FordPass data?  
My utility offers a rebate for 'Off-Peak" charging who's app was refreshing and calling the API every 5 Minutes 24/7 that's 10,080 calls form just one App, before calls from the FordPass app, phone widgets, ect.
Since removing the utilities access, the interval can be attributed to FordPass/widget refresh and the HomeAssistant refresh calls.
![image](https://user-images.githubusercontent.com/98041036/197340263-f87e58d4-4d68-476d-b621-0fae229afda6.png)

**Comment by blaine07** [comment]
For what it’s worth the same happened to me today; my accounts been locked. 

Also, see this:

lots of pages:

https://www.macheforum.com/site/threads/unauthorized-api-use-can-disable-your-account.13893/page-11

**Comment by JVTEAM** [comment]
Also locked out today....

**Comment by itchannel** [comment]
I'd be interested to know if you were using any other apps that use the Fordpass API?

As by default the HA integration only makes a vehicle call every 15mins. Of which that involves at max 5 API requests (Per Vehicle). 

- Call to check oauth token
- Refresh or renew if needed
- Get latest Vehicle status from /status endpoint
- Get latest Messages from /messages

So in theory the maxAPI calls it should be making per week is 3,360 calls (Per Vehicle). which is only 20 per hour which is nothing in API terms.

**Comment by blaine07** [comment]
Link I posted above suggest it isn’t even a number of calls it’s that the request didn’t come from FORD APP. This app is ALL I had setup to do anything with my vehicle. Thanks,Blaine--Sent by carrier pigeonOn Nov 29, 2022, at 15:10, itchannel ***@***.***> wrote:﻿
I'd be interested to know if you were using any other apps that use the Fordpass API?
As by default the HA integration only makes a vehicle call every 15mins. Of which that involves at max 5 API requests (Per Vehicle).

Call to check oauth token
Refresh or renew if needed
Get latest Vehicle status from /status endpoint
Get latest Messages from /messages

So in theory the maxAPI calls it should be making per week is 3,360 calls (Per Vehicle). which is only 20 per hour which is nothing in API terms.

**Comment by itchannel** [comment]
So currently the integration checks every 5mins for status updates. Which shouldn't cause any API issues as that's barely anything but if Ford are making the limit even less then I could add the ability to set your own update interval in the GUI or you can do it manually by editing the __init.py__ files on line 37 and changing the below line to another value in seconds (Keep it more than 300)

```
SCAN_INTERVAL = timedelta(seconds=300)
```

**Comment by itchannel** [comment]
@blaine07 they wanted a physical limit which would be hard to enforce with HA reboots etc. However adding the ability to change how often it updates for each individual user is probably the way forward. 

I've been away for a while but will see what I can mock up later today.

**Comment by itchannel** [comment]
1.39 has been released which includes a new default poll interval of 15mins instead of 5 as well as the ability to change this in options.

**Comment by balthisar** [comment]
Locked out yesterday.

- Nothing else is using this API aside from the FordPass app. I don't use third party services and don't give my credential to anyone. This is the _only_ thing I have accessing the API.
- It's running on my own instance on my own hardware in my own house, and not on an AWS instance, not Nabucasa, etc.
- Possibly related to update intervals, I'll offer that I'm connected to two vehicles with the same credentials. Err, _was_ connected.

**Comment by jrprinty** [comment]
I woke up to being locked out today. 

I do have two vehicles connected through Home Assistant. 

I can't believe Ford would do this to us as a paying customer!

@itchannel Thank you for all your work on this integration! It has served my family well over the years, and I hope Ford can pull their heads together and get this fixed for all of us.

**Comment by serlinGi** [comment]
also to me 2 cars no longer usable with integration; blocked account, error code fordpass app: CAIAH0320E
I was hoping to fix it when I saw that the new update was published 😭

**Comment by gpf2ez** [comment]
Add me to the list of people locked out yesterday. I also have two vehicles that and I got the same email as above and get the same CAIAH0320E error code when I try to login.

For those that have not done so already please call Ford Customer service at 800-392-3673.  They are taking reports and creating tickets.  They person I spoke to said they have been getting a lot of calls.

My guess is Ford recently updated their security policy's and did not take this use case into consideration.

**Comment by jrprinty** [comment]
Well, I just got off the phone with Ford (833-385-0512). They made me delete my FordPass integration from Home Assistant and said it would be 3 to 5 business days before they re-enabled my account. 

I'm so pissed off with Ford right now. Their API is publicly available, but now we can't use it. 

I will set up a burner account once my Ford account is unlocked and see what happens with that account.

**Comment by blaine07** [comment]
> Well, I just got off the phone with Ford ([833-385-0512](tel:833-385-0512)). They made me delete my FordPass integration from Home Assistant and said it would be 3 to 5 business days before they re-enabled my account.
> 
> I'm so pissed off with Ford right now. Their API is publicly available, but now we can't use it.
> 
> I will set up a burner account once my Ford account is unlocked and see what happens with that account.

They just escalated my case two days ago and haven’t heard anything since. Lady asked me if I deleted whatever and I told her yes but she was just who opened ticket 😩

**Comment by jrprinty** [comment]
That's not what I wanted to hear @blaine07 

I sent a public tweet to Ford but haven't gotten a response from them, and I pointed out the FordConnect API. I doubt it will help in any way, but at least it is more public now.

**Comment by itchannel** [comment]
Shame to hear so many people are being locked out. 

What sucks is it uses the documented api calls to the 3rd party API from the documentation on developer.ford.com which I have access to. At no point does it mention any API limitations or rules. I can only guess they want to go down the paywall route.

Hopefully there might be some transparency from Ford by either working with devs to make 3rd party integrations compliant or actual open source tooling from them. More and more are going down the smart home route and integrate their vehicles into other systems like charging, energy monitoring etc :(

**Comment by jrprinty** [comment]
I'm going to bring that up when they call, if they actually call me. 

I am also signed up with a developer account. Hepefully they have a plan that doesn't involve us losing functionality and having a paywall. This is effecting so many people not just in Home Automation but even with saving money with EVs and thier local electric companies.

**Comment by balthisar** [comment]
Say, are any of you using a reverse proxy? Or Nabucasa?

I just had a conversation with an IT Security person with an onshore accent, and she asked if I were accessing the API via a VPN. I offered that I was using a reverse proxy, and she indicated that that may have been one of the flags, because there was a high incidence of IP address changes in the traffic.

I'm not going to start analyzing my traffic, but my HA install sees everything as coming from my xxx.xxx.xxx.9 machine, and transmits all traffic back to that. If there are headers with IP addresses not matching the actual endpoints, that might be what she was referring to.

In any case, she acknowledged that HA is not being locked out intentionally, and that they consider it first party not third party, and that there are people in senior management at Ford that got bitten by this same bug that are anxious to get it solved.

She suggested that the system might not flag at a reduced polling frequency, but couldn't make any guarantees. She was knowledgeable, but has spent her whole day on the phone calling people instead of being part of the team actually solving the problem.

She specifically said it was the FNA IT Security team working on solving the issue, which is comforting.

@itchannel, we're now abusing the issues system instead of reporting actual bugs and working on resolving them. Sorry 'about that. Thanks for letting us communicate here.

**Comment by gpf2ez** [comment]
> Say, are any of you using a reverse proxy? Or Nabucasa?
> 
> I just had a conversation with an IT Security person with an onshore accent, and she asked if I were accessing the API via a VPN. I offered that I was using a reverse proxy, and she indicated that that may have been one of the flags, because there was a high incidence of IP address changes in the traffic.
> 
> I'm not going to start analyzing my traffic, but my HA install sees everything as coming from my xxx.xxx.xxx.9 machine, and transmits all traffic back to that. If there are headers with IP addresses not matching the actual endpoints, that might be what she was referring to.
> 
> In any case, she acknowledged that HA is not being locked out intentionally, and that they consider it first party not third party, and that there are people in senior management at Ford that got bitten by this same bug that are anxious to get it solved.
> 
> She suggested that the system might not flag at a reduced polling frequency, but couldn't make any guarantees. She was knowledgeable, but has spent her whole day on the phone calling people instead of being part of the team actually solving the problem.
> 
> She specifically said it was the FNA IT Security team working on solving the issue, which is comforting.
> 
> @itchannel, we're now abusing the issues system instead of reporting actual bugs and working on resolving them. Sorry 'about that. Thanks for letting us communicate here.

I host from a static IP from my house without a VPN or reverse proxy. I honestly think this comes down to a new employee who decided to look at logs and saw a huge amount of hits through the API and rather than looking to see what's going on just started shutting down accounts.

**Comment by itchannel** [comment]
@balthisar Interesting to hear that they acknowledge HA. Would be nice if someone from Fords dev team reached out so I could work with them to make it nicely compliant and fully supported. We can dream :)  It's either someone from security wanting to limit the amount of exposure or someone from finance who thought "How can we reduce the API request costs lol"

It could potentially be using a VPN but guessing 99% of HA users just have it going out via their own network. I've always tried to limit the amount of information the integration pulls from Ford as I know some others grab internal device details and other hidden functionality which could cause problems and shouldn't really be exposed to the end user but this integration only uses the endpoints from their own 3rd party API docs 👍

**Comment by balthisar** [comment]
Interesting that Home Assistant isn't on that list. The engineer I talked to _definitely_ knew what Home Assistant was.

I wonder what the track is to get this integration out of HACS and into Home Assistant proper? Doing so might lend it a bit more credibility in corporate's eyes.

**Comment by stroodle96** [comment]
Does setting up a second fordpass account for this integration bypass the issue? Or do they lock your account based on vehicle vin#? 

If I create a separate 2nd account and use those credentials for HA integration, then will I still be able to use the Fordpass app with my original account even if the 2nd one gets blocked? Or do they block any fordpass account linked to that vin? 

Can anybody with an actively blocked account try signing up for a new Ford account and link their vehicle vin to see if that works?

**Comment by jrprinty** [comment]
@stroodle96 You cannot set up another account while being locked out without completely abandoning your original Ford account. 

With that said, I have read a couple of people's posts on other forums stating only the account they were using was locked out, and their wife's account was still working without issue.

As soon as my account is unlocked, I will set up a burner account for HA, and I'll see what happens. Hopefully, the reports I have read are correct, and if they disable my burner account, who cares.

**Comment by stroodle96** [comment]
Thank you both for sharing. I thankfully have not been locked out yet. I created a separate shared account to use with this integration and changed the refresh rate to 30 minutes to be extra safe. Hopefully this works to prevent a total lockout of the FordPass app on my my main account.

**Comment by Madbeefer** [comment]
I just called them again since they still haven't unlocked my account from like 2 months ago.. The person I spoke to said she has about 5 people call her a day about their accounts getting locked out..

**Comment by GeneralLouis** [comment]
Got a definite answer on why my account was locked: Cyber security team checked the logs and my account swapped IPs 1500 times in a 24 hour period so the disabled my account.

The FordPass help desk can enable peoples accounts again.

What I'm confused about is how that happened? Would the Nabu Casa cloud cause such a thing with outbound traffic?

**Comment by GeneralLouis** [comment]
Sorry should have clarified, the FordPass help desk is the way to get your ticket submitted to the internal team to enable the account.

**Comment by blaine07** [comment]
I still don’t have my account unlocked. Facepalm

**Comment by jrprinty** [comment]
@blaine07 If you have social media, start annoying them on there. 

They kept telling me I had to wait and that they would get back to me, so I started bugging them on Twitter(@Ford), and once someone else commented on my tweet, they were very quick to get my account unlocked. From the time they told me to call in to getting my account unlocked was about three hours, but I started the process 7 days ago. 

I don't know if the Twitter thing helps or if it was a coincidence, but it can't hurt. I know you have been waiting a lot longer than I did.

**Comment by gpf2ez** [comment]
> @GeneralLouis I think the rep misspoke or was blowing smoke. I can't see any possible way your IP would change that many times in 24 hours. But, of course, that is just my opinion.
> 
> I finally got my account unlocked and set up a burner account to use in Home Assistant. I set up two vehicles and set the refresh rate to 30 minutes. We will see what happens.
> 
> The lady I was talking to wasn't very helpful at all. I explained multiple times we were using this integration within Home Assistant, and it follows all of their API guidelines for FordConnect API, but she just couldn't grasp that concept. I also informed her that I have a developer account with them and asked if they had changed their guidelines or have new recommendations to please send them, but she said she hadn't heard anything about this. She ultimately told me that the API had to have made hundreds of API requests, and their system found them fraudulent, which we all know it didn't.
> 
> She also informed me that they are dealing with a lot of calls about accounts being locked out, and their cyber security team is looking at changing the way it works, and hopefully, it will be less restrictive in the future. I'm not holding my breath, though.
> 
> Like I said not very helpful, but I thought I would share my experience with you guys.

From your description it really sounds like they implemented a new security software or system and it started flagging because they didn't have their rules set up to match their API. And now they're dealing with the fallout from that. 
Did they call you directly to talk to you and get your account unlocked? I was told mine would be unlocked by today but I haven't received any correspondence from them about it at all.

**Comment by GeneralLouis** [comment]
@jrprinty , my personal theory is, after reading the code here, that every single HA user has the same client ID, so if cyber security was simply tracking Client ID they would see TONS of calls coming from all over the world (every HA instance with Fordpass integration) since this implementation has a static client ID coded in. (I have gone in and changed mine and it appears to be working fine on my new burner account after getting unlocked) I personally reached out to team members on the cyber security team and they looked at the logs and got back to me, though I have not heard back when I asked for details, this was not a phone/support agent.

@gpf2ez , I called last week to the Ford support number AND emailed an internal contact, so I had 2 tickets going basically. Yesterday I called the Fordpass support line instead, they told me the previous call was in a different workstream and they would get a new ticket in since I had not heard back, within 3 hours I got the generic IBM email that said "Account : Enabled", so I am not entirely sure which "ticket" caused it to go through, I have had no follow up from any source.

When any of you with disabled accounts call and they ask about Third-party use, make sure you tell them, you are using the APIs yourself. Since this code is open source and you are running it locally, your credentials are not being "given" to anyone else, you are simply using code written to help you use their API's, something you could do yourself if you read their API documentation. (and had TONS of time on your hand to figure it out like the author(s) of this code have done)

**Comment by itchannel** [comment]
@GeneralLouis I would be very careful changing Client ID's especially ones that are not approved. A Client ID is per application not per user. The current Client ID is the one used for Fordpass as they haven't currently issued specific ones for 3rd party applications that I am aware of.

**Comment by GeneralLouis** [comment]
@itchannel , thanks I will revert my changes, as I did not fully understand client ID and it's use.

**Comment by JVTEAM** [comment]
I need to call them back today as it's been a week.  I wrote down my case number, but it's among other numbers that I didn't label.  Does it start with cas-?

**Comment by GeneralLouis** [comment]
@JVTEAM yes, they start with cas-. I was told to call 800-336-0486 to get to the correct queue if I had to call back. Double-check their hours, they may still be closed for 30 minutes.

**Comment by blaine07** [comment]
Yeah I need to follow up on mine getting unlocked but no idea how to get a hold of someone without being on hold for stupid days.

**Comment by jrprinty** [comment]
@blaine07 I called ‭+1 (833) 385-0512‬ , and I think it was option 2 to talk to a FordPass Guide. It only took a few minutes before someone picked up.

@gpf2ez I'm glad to hear you got your account unlocked. I wish this process was more streamlined, but at the same time, it shouldn't have happened in the first place.

I agree completely with them not having the rules set up properly when they implemented this new security software. I know things happen, and they are ultimately trying to protect our accounts which I appreciate. Hopefully, in the future, they will have this all squared away, and we, as the Ford community, will not have to deal with account lock-outs and still be able to use the API.

**Comment by blaine07** [comment]
> @gpf2ez I agree completely with them not having the rules set up properly when they implemented this new security software. I know things happen, and they are ultimately trying to protect our accounts which I appreciate. Hopefully, in the future, they will have this all squared away, and we, as the Ford community, will not have to deal with account lock-outs and still be able to use the API.
> 
> @blaine07 I called ‭+1 (833) 385-0512‬ , and I think it was option 2 to talk to a FordPass Guide. It only took a few minutes before someone picked up.
> 
> @gpf2ez I'm glad to hear you got your account unlocked. I wish this process was more streamlined, but at the same time, it shouldn't have happened in the first place.

Called again, guy was very helpful. Said someone should be calling me today :cross fingers:

**Comment by itchannel** [comment]
@cjramseyer User-Agent is set on line 19 in "fordpass_new.py" you are welcome to change it to another value however by default I have it set to the Fordpass application User-Agent.

**Comment by blaine07** [comment]
> > @gpf2ez I agree completely with them not having the rules set up properly when they implemented this new security software. I know things happen, and they are ultimately trying to protect our accounts which I appreciate. Hopefully, in the future, they will have this all squared away, and we, as the Ford community, will not have to deal with account lock-outs and still be able to use the API.
> > @blaine07 I called ‭+1 (833) 385-0512‬ , and I think it was option 2 to talk to a FordPass Guide. It only took a few minutes before someone picked up.
> > @gpf2ez I'm glad to hear you got your account unlocked. I wish this process was more streamlined, but at the same time, it shouldn't have happened in the first place.
> 
> Called again, guy was very helpful. Said someone should be calling me today :cross fingers:

About a hour later feller got me sorted out. Something about 3000 tickets in last 30 days or something. Very polite but seemed exhausted lol

**Comment by dmkjr** [comment]
I created my ticket last night. I had the rep add to the case that I was looking at API call information in regards to at what point their tools trip as "fraud". Not hopeful for a response, but we will see.

**Comment by JVTEAM** [comment]
Well...It's been 2 weeks.  guess I should call them back.  What a nightmare.

**Comment by boctok** [comment]
I would really like to have this in my HA, but I also don't want to get locked out of my app. Is there any surefire way to use this, limiting the API calls so that I won't bring down the wrath of the Fordpass gods?

**Comment by GeneralLouis** [comment]
@boctok I am trying out alt FordPass accounts and 30-minute refresh in HA, I will see what happens. I am keeping my Fordpass main account with Google Assistant for now, mainly because I can't figure out how to remove it (lol)

@JVTEAM I got the same message from the cyber security team when they checked the logs for me, something odd must be happening because IDK how so many people could have done the same thing. Maybe a data breach or a testing unit that should have been in QA or Dev hit Prod and hammered the API giving false positives on everyone's accounts?

**Comment by JVTEAM** [comment]
> @boctok I am trying out alt FordPass accounts and 30-minute refresh in HA, I will see what happens. I am keeping my Fordpass main account with Google Assistant for now, mainly because I can't figure out how to remove it (lol)
> 
> @JVTEAM I got the same message from the cyber security team when they checked the logs for me, something odd must be happening because IDK how so many people could have done the same thing. Maybe a data breach or a testing unit that should have been in QA or Dev hit Prod and hammered the API giving false positives on everyone's accounts?

I have quite a few devices accessing home assistant. I don't know if the API call comes from the HA server or the device, but I believe it's the client device. That being said they also said "changed' not new IP address.  So I could see my phone switching from wifi to cell, then onto a VPN, then off the vpn, then off the wifi and all the other devices pinging it at the same time.  So perhaps it's more about elminating some of the devices that are using it and centrailzing any info I need from it to a source that doesn't care about my IP.

**Comment by dmkjr** [comment]
I just talked to Ford again this morning.  It appears that you must call back in after the "waiting period".  He was able to speak directly to one of the folks in between customer support and engineering.  Waiting on email now about it being unlocked.  I asked several questions to append to the case to see if we can get some resolution.

Someone else said their tech sounded exhausted.  This gentlemen said every other call is about someone being locked out of their account and has been like this for weeks.  Ford screwed up.

**Comment by dmkjr** [comment]
> I just talked to Ford again this morning. It appears that you must call back in after the "waiting period". He was able to speak directly to one of the folks in between customer support and engineering. Waiting on email now about it being unlocked. I asked several questions to append to the case to see if we can get some resolution.
> 
> Someone else said their tech sounded exhausted. This gentlemen said every other call is about someone being locked out of their account and has been like this for weeks. Ford screwed up.

Quick update -- account is still not unlocked.  Waiting for the customer service department to open so I can, again, call back.

**Comment by michalk-k** [comment]
> 



> I have quite a few devices accessing home assistant. I don't know if the API call comes from the HA server or the device, but I believe it's the client device.

In the case of HA, it's your HA server calling their API. Then what Ford server sees depends on how you have arranged a connection from HA to the Internet

**Comment by mcrummett** [comment]
> @boctok I am trying out alt FordPass accounts and 30-minute refresh in HA, I will see what happens. I am keeping my Fordpass main account with Google Assistant for now, mainly because I can't figure out how to remove it (lol)
> 
> @JVTEAM I got the same message from the cyber security team when they checked the logs for me, something odd must be happening because IDK how so many people could have done the same thing. Maybe a data breach or a testing unit that should have been in QA or Dev hit Prod and hammered the API giving false positives on everyone's accounts?

How did you get on with this in the end with 30 minute refresh in HA RE account lockouts?

**Comment by itchannel** [comment]
I haven't had any reports of lockouts since making the change. However there is always a small risk given this is not an official supported integration by Ford. As much as I wish they would either work with me or bring out one of their own :)

**Comment by mcrummett** [comment]
> I haven't had any reports of lockouts since making the change. However there is always a small risk given this is not an official supported integration by Ford. As much as I wish they would either work with me or bring out one of their own :)

Legend. Thanks for your work on this.

**Comment by Ho0m3s** [comment]
hello, I have been trying to use this integration with my Ford Edge 2017 andHomeAssistant. I get the following error messages. i am not locked out or anything, but i am not sure why it is not connecting. the Vin is correct and the email(redacted). ideas or thoughts would be welcome :) 

```
2023-07-23 20:06:27.702 WARNING (MainThread) [homeassistant.config_entries] Config entry 'Vehicle (XXXXXXXXXXXXXXXX)' for fordpass integration not ready yet: None; Retrying in background
2023-07-23 20:06:33.179 DEBUG (MainThread) [custom_components.fordpass] 300000
2023-07-23 20:06:33.179 DEBUG (MainThread) [custom_components.fordpass] username
2023-07-23 20:06:33.179 DEBUG (MainThread) [custom_components.fordpass] password
2023-07-23 20:06:33.179 DEBUG (MainThread) [custom_components.fordpass] vin
2023-07-23 20:06:33.180 DEBUG (MainThread) [custom_components.fordpass] region
2023-07-23 20:06:33.180 DEBUG (MainThread) [custom_components.fordpass] North America & Canada
2023-07-23 20:06:33.193 DEBUG (MainThread) [custom_components.fordpass.fordpass_new] /config/custom_components/fordpass/valid-email@gmail.com_fordpass_token.txt
2023-07-23 20:06:33.196 DEBUG (SyncWorker_6) [custom_components.fordpass.fordpass_new] Token is valid, continuing
2023-07-23 20:06:33.802 DEBUG (SyncWorker_6) [custom_components.fordpass.fordpass_new] 401 with status request: start token refresh
2023-07-23 20:06:34.184 DEBUG (SyncWorker_6) [custom_components.fordpass.fordpass_new] Token is valid, continuing
2023-07-23 20:06:34.444 WARNING (MainThread) [custom_components.fordpass] 'vehiclestatus'
2023-07-23 20:06:34.448 WARNING (MainThread) [custom_components.fordpass] Error communicating with FordPass for XXXXXXXXXXXXXXXX
2023-07-23 20:06:34.451 ERROR (MainThread) [custom_components.fordpass] Error fetching fordpass data: Error communicating with FordPass for XXXXXXXXXXXXXXXX
2023-07-23 20:06:34.453 DEBUG (MainThread) [custom_components.fordpass] Finished fetching fordpass data in 1.260 seconds (success: False)
`

**Comment by itchannel** [comment]
> hello, I have been trying to use this integration with my Ford Edge 2017 andHomeAssistant. I get the following error messages. i am not locked out or anything, but i am not sure why it is not connecting. the Vin is correct and the email(redacted). ideas or thoughts would be welcome :)
> 
> ```
> 2023-07-23 20:06:27.702 WARNING (MainThread) [homeassistant.config_entries] Config entry 'Vehicle (XXXXXXXXXXXXXXXX)' for fordpass integration not ready yet: None; Retrying in background
> 2023-07-23 20:06:33.179 DEBUG (MainThread) [custom_components.fordpass] 300000
> 2023-07-23 20:06:33.179 DEBUG (MainThread) [custom_components.fordpass] username
> 2023-07-23 20:06:33.179 DEBUG (MainThread) [custom_components.fordpass] password
> 2023-07-23 20:06:33.179 DEBUG (MainThread) [custom_components.fordpass] vin
> 2023-07-23 20:06:33.180 DEBUG (MainThread) [custom_components.fordpass] region
> 2023-07-23 20:06:33.180 DEBUG (MainThread) [custom_components.fordpass] North America & Canada
> 2023-07-23 20:06:33.193 DEBUG (MainThread) [custom_components.fordpass.fordpass_new] /config/custom_components/fordpass/valid-email@gmail.com_fordpass_token.txt
> 2023-07-23 20:06:33.196 DEBUG (SyncWorker_6) [custom_components.fordpass.fordpass_new] Token is valid, continuing
> 2023-07-23 20:06:33.802 DEBUG (SyncWorker_6) [custom_components.fordpass.fordpass_new] 401 with status request: start token refresh
> 2023-07-23 20:06:34.184 DEBUG (SyncWorker_6) [custom_components.fordpass.fordpass_new] Token is valid, continuing
> 2023-07-23 20:06:34.444 WARNING (MainThread) [custom_components.fordpass] 'vehiclestatus'
> 2023-07-23 20:06:34.448 WARNING (MainThread) [custom_components.fordpass] Error communicating with FordPass for XXXXXXXXXXXXXXXX
> 2023-07-23 20:06:34.451 ERROR (MainThread) [custom_components.fordpass] Error fetching fordpass data: Error communicating with FordPass for XXXXXXXXXXXXXXXX
> 2023-07-23 20:06:34.453 DEBUG (MainThread) [custom_components.fordpass] Finished fetching fordpass data in 1.260 seconds (success: False)
> `
> ```

Looking at the log output it appears it was unable to return any info for your vehicle:

```
2023-07-23 20:06:34.444 WARNING (MainThread) [custom_components.fordpass] 'vehiclestatus'
```

Is your car supported in the Fordpass application?

If so then your login details are correct but your VIN isn't so might be worth double checking.

**Comment by t785634** [comment]
Is it still a problem that ford blocks your account?

**Comment by mcrummett** [comment]
Been all good for a long time now :). The reduction in API calls seemed to sort it

Mark Crummett
________________________________

**Comment by gee-jay-bee** [comment]
My account was locked out this week. I have no idea why as I have been running this integration for months without issue on a dedicated account and the only activity would be driven by HA polling! I will call Ford later to reactivate and see if I can get more info. Out of interest, can I suppress polling and use homeassistant.update_entity service to control refresh for myself? I generally have a good idea when to refresh from other sources so can perhaps reduce the calls to a bare minimum
